shared/modules/Numeric.ts

```diff
@@ -0,0 +1,621 @@
+import { BigNumber } from 'bignumber.js';
```

Comment:
Would it be possible to break this PR up into stages, to make it easier to review?

e.g.
* Add this new `Numeric` module
* Migrating code from `conversion.utils.js` to `Numeric` in chunks
* Delete `conversion.utils.js`

That might make this less intimidating to review. And we can more effectively assign each chunk to the people most familiar with that code (e.g. we could ask the swaps team to just review the swaps changes).

Reply:
@Gudahtt I will break this down *some* but I don't know that I can break it down to the level of granularity that would be the easiest to review. I say this for the sake of expediency with hitting our goals for Q1 within the allotted time we have for TS as well because there are some order of operations issues. @DDDDDanica also asked for this, so cc'ing her for acknowledgement and thanking her for pushing back as well. 

> Add this new Numeric module

This would be easy to split out, considering it wouldn't be used anywhere yet.

> Migrating code from conversion.utils.js to Numeric in chunks

So to do this I would need to redo some things because I converted conversion.utils.js to typescript *first* to get type feedback. To be able to merge this in chunks I would need to migrate to Numeric in chunks in *JavaScript* because if i attempt to do this in TypeScript it will not be mergeable due to lint / compile errors. So this would involve more rework than I think if sensible with the time constraints mentioned. However...

There's the fact that I relocated a bunch of methods that *use* conversionUtil from other parts of the codebase to the conversion.utils.js file. That would be a more meaningful break-up strategy and would mitigate a lot of LOC from this PR. I already did this PR breakdown here: https://github.com/MetaMask/metamask-extension/pull/17319

I will see what else can be shaved off into separate PRs.

Reply:
Oh sorry, I explained that poorly. I meant to say:

>Migrating code that _uses  `conversion.utils.js` to use `Numeric` instead, in chunks

Reply:
@Gudahtt I think i've got it fairly granular now. Some things have already merged, namely Numeric module and the relocation of various helper methods to the conversion.utils.js shared file. I will rename this PR and move from draft when it comes up but it now only contains the changes from the conversion.utils.js -> conversion.utils.ts conversion.